x/concentrated-liquidity/incentives_test.go

```diff
@@ -0,0 +1,119 @@
+package concentrated_liquidity_test
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	cl "github.com/osmosis-labs/osmosis/v14/x/concentrated-liquidity"
+	"github.com/osmosis-labs/osmosis/v14/x/concentrated-liquidity/types"
+)
+
+var defaultPoolId = uint64(1)
+
+func (s *KeeperTestSuite) TestCreateAndGetUptimeAccumulators() {
+	// We expect there to be len(types.SupportedUptimes) number of initialized accumulators
+	// for a successful pool creation. We calculate this upfront to ensure test compatibility
+	// if the uptimes we support ever change.
+	curExpectedAccumValues := []sdk.DecCoins{}
+	for range types.SupportedUptimes {
+		curExpectedAccumValues = append(curExpectedAccumValues, cl.EmptyCoins)
+	}
+	s.Require().Equal(len(types.SupportedUptimes), len(curExpectedAccumValues))
+
+	type initUptimeAccumTest struct {
+		poolId              uint64
+		initializePool      bool
+		expectedAccumValues []sdk.DecCoins
+
+		expectedPass bool
+	}
+	tests := map[string]initUptimeAccumTest{
+		"default pool setup": {
+			poolId:              defaultPoolId,
+			initializePool:      true,
+			expectedAccumValues: curExpectedAccumValues,
+			expectedPass:        true,
+		},
+		"setup with different poolId": {
+			poolId:              defaultPoolId + 1,
+			initializePool:      true,
+			expectedAccumValues: curExpectedAccumValues,
+			expectedPass:        true,
+		},
+		"pool not initialized": {
+			initializePool:      false,
+			poolId:              defaultPoolId,
+			expectedAccumValues: []sdk.DecCoins{},
+			expectedPass:        false,
+		},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		s.Run(name, func() {
+			s.SetupTest()
+			clKeeper := s.App.ConcentratedLiquidityKeeper
+
+			// system under test
+			var err error
+			if tc.initializePool {
+				err = clKeeper.CreateUptimeAccumulators(s.Ctx, tc.poolId)
```

Comment:
should error check here since the err gets overwritten by GetUptimeAccumulators by the time we check it

Reply:
Nice catch, fixed